web/react/components/user_settings/manage_incoming_hooks.jsx

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2015 Spinpunch, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+var Client = require('../../utils/client.jsx');
+var Utils = require('../../utils/utils.jsx');
+var Constants = require('../../utils/constants.jsx');
+var ChannelStore = require('../../stores/channel_store.jsx');
+var LoadingScreen = require('../loading_screen.jsx');
+
+export default class ManageIncomingHooks extends React.Component {
+    constructor() {
+        super();
+
+        this.getHooks = this.getHooks.bind(this);
+        this.addNewHook = this.addNewHook.bind(this);
+        this.updateChannelId = this.updateChannelId.bind(this);
+
+        this.state = {hooks: [], channelId: ChannelStore.getByName(Constants.DEFAULT_CHANNEL).id, getHooksComplete: false};
+    }
+    componentDidMount() {
+        this.getHooks();
+    }
+    addNewHook() {
+        let hook = {}; //eslint-disable-line prefer-const
```

Comment:
Yes const is stupid in javascript. However we are using ESLint which should force the immutability of const variables. (rule: no-const-assign)  If the rule had to be disabled here it is probably a bug in ESLint causing it, as it is modified below.